### DISCUSSION TITLE
Moving _site files to CPanel File Manager: folders won't upload

### DISCUSSION THREAD

**Discussion by hnswans** [CollectionBuilder/collectionbuilder.github.io#57] [Q&A]
Hi there,

I'm new to this and trying to test out deploying my site.

After building in Jekyll, I'm trying to move all of _site across to CPanel's File Manager, but I get error messages from CPanel that I can't upload folders, only files.

Here's the repository: [https://github.com/hnswans/simsproj2022](https://github.com/hnswans/simsproj2022)

And here's what the site looks like right now, so obviously something's not working: [http://movingstories.hnswanson.upenn.domains/](http://movingstories.hnswanson.upenn.domains/)

What should I be doing? Thanks so much in advance for your help!

**Comment by evanwill** [comment]
@hnswans unfortunately cPanel File Manager doesn't allow you to upload a complete folder of files, so that is the error you are encountering, and your site currently has nothing in it (thus the empty default you see at your URL currently!). 

Generally, the best option is to set up FTP access for yourself--see [cPanel FTP account docs](https://docs.cpanel.net/cpanel/files/ftp-accounts/) or your org's [upenn docs](https://domains.library.upenn.edu/support/#setting-up-ftp). With FTP you will use a FTP client to sync the contents of the "_site" folder from your local machine to your "public_html" folder on your host (correct destination might be different since you are using a subdomain). It should be pretty easy to transfer files and update them once you have it set up. 

If you can't get FTP set up, here is a workaround:
On your local computer zip the "_site" folder into a .zip file. Then use File Manager to upload the .zip. Then on File Manager use the "Extract" button to extract the contents of the "_site.zip" into "public_html" (or appropriate location).

**Reply by evanwill** [reply]
@hnswans the answer above is specifically about setting up FTP to manually deploy when you are building from your local machine. 
Looking at your repository, it seems like you have a GitHub Action set up to build and deploy on GitHub Pages ([jekyll.yml file](https://github.com/hnswans/simsproj2022/blob/main/.github/workflows/jekyll.yml)). 

If you want to keep using your GitHub Action built version, you can use [GitHub Pages custom domain option](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site) (rather than building your site manually and moving the files to your server, you just let GitHub Pages keep hosting the site, but put your custom domain URL in front of it).

If you want to switch to manually building your site and uploading to your server space (as described above), then you should delete the GitHub Action [jekyll.yml file](https://github.com/hnswans/simsproj2022/blob/main/.github/workflows/jekyll.yml) file from your project and turn off GitHub Pages.

I am not sure what your [.cpanel.yml file](https://github.com/hnswans/simsproj2022/blob/main/.cpanel.yml) is doing--it looks like trying to set up an automated deployment, but the content seems like a Git Merge conflict messed up the file and it isn't cleaned up--so might be another issue to sort out (or maybe just delete the file!).
